src/app-layer-htp-file.c

```diff
@@ -161,60 +161,10 @@ int HTPFileOpen(HtpState *s, HtpTxUserData *tx, const uint8_t *filename, uint16_
  *
  * @return HTP_OK on success, HTP_ERROR on failure.
  */
-int HTPParseContentRange(bstr * rawvalue, HtpContentRange *range)
+int HTPParseContentRange(bstr *rawvalue, HTTPContentRange *range)
 {
-    unsigned char *data = bstr_ptr(rawvalue);
-    size_t len = bstr_len(rawvalue);
-    size_t pos = 0;
-    size_t last_pos;
-
-    // skip spaces and units
-    while (pos < len && data[pos] == ' ')
-        pos++;
-    while (pos < len && data[pos] != ' ')
-        pos++;
-    while (pos < len && data[pos] == ' ')
-        pos++;
-
-    // initialize to unseen
-    range->start = -1;
-    range->end = -1;
-    range->size = -1;
-
-    if (pos == len) {
-        // missing data
-        return -1;
-    }
-
-    if (data[pos] == '*') {
-        // case with size only
-        if (len <= pos + 1 || data[pos+1] != '/') {
-            range->size = -1;
-            return -1;
-        }
-        pos += 2;
-        range->size = bstr_util_mem_to_pint(data + pos, len - pos, 10, &last_pos);
-    } else {
-        // case with start and end
-        range->start = bstr_util_mem_to_pint(data + pos, len - pos, 10, &last_pos);
-        pos += last_pos;
-        if (len <= pos + 1 || data[pos] != '-') {
-            return -1;
-        }
-        pos++;
-        range->end = bstr_util_mem_to_pint(data + pos, len - pos, 10, &last_pos);
-        pos += last_pos;
-        if (len <= pos + 1 || data[pos] != '/') {
-            return -1;
-        }
-        pos++;
-        if (data[pos] != '*') {
-            // case with size
-            range->size = bstr_util_mem_to_pint(data + pos, len - pos, 10, &last_pos);
-        }
-    }
-
-    return 0;
+    uint32_t len = bstr_len(rawvalue);
+    return rs_http2_parse_content_range(range, bstr_ptr(rawvalue), len);
```

Comment:
nit: I don't like that the call has 'http2' in the name

Reply:
indeed, some leftover from development version when there were 2 parsing functions